### ISSUE TITLE
Ability to see views of the blog page

### ISSUE THREAD

**Issue by poanchen** [poanchen/poanchen.github.io#21]
We already use Google Analytics for tracking page performance. We should be able to leverage that. Did some research, we could do this, https://stackoverflow.com/questions/19484009/google-analytics-api-get-page-view-information-for-specific-urls

If we are making too many requests, we need to think about how to minimize it. Also, we don't need up to date data. So, maybe we can store the data somewhere and update it once every hour or sth like that.

**Comment by poanchen** [comment]
Only when on the blog page, we will be querying views for all the posts on the site.  We would need optimization here. But let's take baby step and make individual post to have views first. I feel less people go to blog page but quite a few of people go to a post from search engine. Let's do individual post view first

**Comment by poanchen** [comment]
Exactly what I did. Stored it as JSON in Azure Blob Storage. Update the data once every day. Getting the JSON from the Azure Blob Storage is easy in browser.

Here is the commit for client side code: https://github.com/poanchen/poanchen.github.io/commit/b09152e040a14dfdec57fdb50e042274e12e9a93

Here is the commit for saving it to Azure Blob Storage along with parsing, getting data from google apis: https://github.com/poanchen/upload-post-stats

**Comment by poanchen** [comment]
each post got views

**Comment by poanchen** [comment]
- We do leverage the google core reporting api to get views
- However, after research, I found out that in order for it to work on browser, you would need to have authentication with sign in which doesn't work in our case. As we need this data public. So we went with server side approach where we have secret store in server side.
- we also minimize the request call to google reporting api as we don't need to have up to date data. One day old data should be fine to start with. However, there are still as many calls to Azure Blob Storage but since the json file is small. Should be fine. 
- one thing is that, currently even one post still queries all of the data. Maybe we need to use single Asa for azure blob storage and then change the file name to only that blog post whenever we call. But we will end up a lot or files comparing with one big file. Optimization that we can think of for future. 
- currently each post would have views for their own. Blog page does not show any views yet.
